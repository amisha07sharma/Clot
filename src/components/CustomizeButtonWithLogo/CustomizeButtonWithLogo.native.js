import React from "react";
import PropTypes from "prop-types";
import { Image } from "react-native";
import {
  Wrapper,
  TextWrapper,
  TextContainer,
} from "./CustomizeButtonWithLogo.style";

const CustomizeButtonWithLogo = ({
  imageSource,
  buttonText,
  onPress = () => {},
}) => {
  return (
    <Wrapper onPress={onPress} testID="Press">
      <Image
        source={imageSource}
        style={{ height: 40, width: 40, borderRadius: 200 }}
      />
      <TextContainer>
        <TextWrapper>{buttonText}</TextWrapper>
      </TextContainer>
    </Wrapper>
  );
};

CustomizeButtonWithLogo.prototype = {
  imageSource: PropTypes.string.isRequired,
  buttonText: PropTypes.string.isRequired,
  onPress: PropTypes.func,
};
export default CustomizeButtonWithLogo;
