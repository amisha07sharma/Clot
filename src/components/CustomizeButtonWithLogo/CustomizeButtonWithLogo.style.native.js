import styled from "styled-components/native";
import { Text, TouchableOpacity } from "react-native";
import { LIGHT_GREY, BLACK_100 } from "../../constants/colors";

export const Wrapper = styled(TouchableOpacity)`
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
  background-color: ${LIGHT_GREY};
  padding: 12px;
  margin-bottom: 10px;
  border-radius: 100px;
`;

export const TextContainer = styled.View`
  flex: 0.8;
`;

export const TextWrapper = styled(Text)`
  font-size: 16px;
  font-weight: 500;
  line-height: 26.73px;
  letter-spacing: -0.4957527816295624px;
  text-align: center;
  color: ${BLACK_100};
`;
