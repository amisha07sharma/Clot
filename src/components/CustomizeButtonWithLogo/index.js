import CustomizeButtonWithLogo from "./CustomizeButtonWithLogo";

export default CustomizeButtonWithLogo;
