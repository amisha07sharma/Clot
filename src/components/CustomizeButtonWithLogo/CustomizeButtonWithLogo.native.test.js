import React from "react";
import { render, fireEvent, screen } from "@testing-library/react-native";
import CustomizeButtonWithLogo from "./CustomizeButtonWithLogo";

const mockOnPress = jest.fn();
const mockProps = {
  imageSource: "someImageURL",
  buttonText: "Button",
  onPress: mockOnPress,
};

const renderComponent = (props) => {
  render(<CustomizeButtonWithLogo {...mockProps} {...props} />);
};
describe("CustomizeButtonWithLogo component", () => {
  it("renders correctly with default props", () => {
    const wrapper = renderComponent();
    expect(wrapper).toMatchSnapshot();
  });

  it("handles onPress event correctly", () => {
    renderComponent();
    fireEvent(screen.getByTestId("Press"), "onPress");
    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });
});
