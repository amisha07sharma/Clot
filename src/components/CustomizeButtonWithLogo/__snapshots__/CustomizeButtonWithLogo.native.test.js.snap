// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomizeButtonWithLogo component renders correctly with default props 1`] = `undefined`;
